media/js/firefox/new/common/thanks-direct.js

```diff
@@ -92,7 +92,12 @@
         Mozilla.StubAttribution.waitForGoogleAnalytics(function () {
             var data = Mozilla.StubAttribution.getAttributionData();
 
-            if (data && Mozilla.StubAttribution.withinAttributionRate()) {
+            // make sure we check referrer for AMO (issue 11467)
+            if (
+                data &&
+                Mozilla.StubAttribution.withinAttributionRate() &&
+                Mozilla.StubAttribution.hasValidData(data)
+            ) {
```

Comment:
What about other download pages ? If you directly hit (without coming from AMO) a different download page that doesn't call `hasValidData()`, but still passing the AMO specific query parameters, could you bypass the referer check entirely and still get attribution set on the executable ?

Reply:
@diox if I understand the bug report, this is only about the special `/thanks/?s=direct` page. This check is already present in the regular stub attribution flow.